.github/workflows/on-push-to-main.yml

```diff
@@ -94,3 +43,32 @@ jobs:
           template_file: ./README.template.md
           output_file: ./README.md
           dev_docs_slug: rust
+  
+  release-please:
+    name: Release Please
+    runs-on: ubuntu-latest
+    needs: [ generate_readme ]
+    outputs:
+      release_created: ${{ steps.release.outputs.release_created }}
+    steps:
+      - uses: google-github-actions/release-please-action@v3
+        id: release
+        with:
+          token: ${{ secrets.MOMENTO_MACHINE_USER_GITHUB_TOKEN }}
+          release-type: rust
+          package-name: momento
+          default-branch: main
+          changelog-types: '[{"type":"feat","section":"Features","hidden":false},{"type":"fix","section":"Bug Fixes","hidden":false},{"type":"chore","section":"Miscellaneous","hidden":false}]'
+          extra-files: | 
+            Cargo.toml
+
+  publish:
+    name: Publish to crates.io
+    runs-on: ubuntu-latest
+    needs: [ release-please ]
+    if: ${{ needs.release-please.outputs.release_created == 'true' }}
+    steps:
+      - run: |
+          cargo publish
+          --token ${{ secrets.CARGO_REGISTRY_TOKEN }}
```

Comment:
Found this [forum post](https://users.rust-lang.org/t/does-anyone-use-github-actions-to-run-cargo-publish/92374/5) that provides token using this `--token` argument instead of doing a `cargo login` step.

Not sure if I need to include `- uses: actions/checkout@v3` as a step before publishing

Reply:
Or I can try to use this github action for publishing crates instead? https://github.com/marketplace/actions/publish-crates

Reply:
i'm fine with trying what you have above. the stuff we had before was working so probably sticking closer to that will be less disruptive than changing to a new plugin